docs/docs/contribute/software/add-new-metric-provider.md

```diff
@@ -0,0 +1,100 @@
+  `EvaluateQuery`(Fetches metric values from the provider)
+   - This function fetches metric values based on the provided
+     metric query from the provider.
+     It evaluates the query and returns the metric values
+     along with any additional data if required.
+  `EvaluateQueryForStep`(Fetches metric values with step interval from the provider)
+   - This function fetches metric values with a specified step interval from the dummy provider.
+      It takes into account the metric query and the step interval provided, executes the query,
+      and returns the metric values along with any additional data if required.
+  `FetchAnalysisValue`(Fetches analysis values from the provider) functions.
+   - This function fetches analysis values based on the provided query and time range from the
+     provider.
+     It evaluates the query within the specified time range and returns the analysis
+     values along with any additional data if required.
+  You can follow other existing implementations,
+ such as [prometheus.go](https://github.com/keptn/lifecycle-toolkit/blob/main/metrics-operator/controllers/common/providers/prometheus/prometheus.go),
+ as an example.
+  Below is an example of a dummy provider implementation.
+
+    ```go
+    {% include "./dummy-code-example.go" %}
+    ```
+
+   **NB:** Ensure to refer to the documentation of
+    [metrics](https://github.com/keptn/lifecycle-toolkit/blob/main/docs/docs/reference/crd-reference/metric.md)
+    and [analysis](https://github.com/keptn/lifecycle-toolkit/blob/main/docs/docs/reference/crd-reference/analysis.md)
+    to understand what data should be retrieved from the methods inputs to compute accurate results.
+
+4. **Instantiate the Provider:** In the `providers.NewProvider` function
+ in the `metrics-operator/controllers/common/providers/provider.go` file,
+ add a case for the `KeptnDummyProviderType`.
+  Instantiate the dummy provider struct and return it.
+
+    ```go
+    // Inside the providers package
+
+    // NewProvider function
+    func NewProvider(providerType string, log logr.Logger, k8sClient client.Client) (KeptnSLIProvider, error) {
+        switch strings.ToLower(providerType) {
+        case KeptnDummyProviderType:
+            return &dummy.KeptnDummyProvider{
+                Log:        log,
+                HttpClient: http.Client{},
+            }, nil
+        // Other cases...
+        }
+    }
+    ```
+
+5. **Update validation webhook and crd config:** To update the validation webhook and crd config of the metrics operator.
+
+   Add the provider name next to last provider in this
+   [line](https://github.com/keptn/lifecycle-toolkit/blob/main/metrics-operator/api/v1beta1/keptnmetricsprovider_types.go#L29)
+   to look like this `// +kubebuilder:validation:Pattern:=prometheus|dynatrace|datadog|dql|dummy`.
+
+   In the metric-operator directory run `make manifests` to update the metrics-operator crd config
```

Comment:
you forgot to mention to modify the helm chart crd  validation

Reply:
I did I will add it to the step.